### PR TITLE
Onboarding RecoveryServices for autogeneration

### PR DESCRIPTION
**Pull request by leni-msft**
```
> azure-schema-generator@1.0.0 generate-single /home/leni/workspace/azure-resource-manager-schemas/generator
> ts-node cmd/generatesingle "recoveryservices/resource-manager"

[/tmp/schm_azspc] executing: git fsck
[/tmp/schm_azspc] executing: git reset -q .
[/tmp/schm_azspc] executing: git checkout -q -- .
[/tmp/schm_azspc] executing: git clean -q -fd
[/tmp/schm_azspc] executing: git gc
[/tmp/schm_azspc] executing: git fetch -q
[/tmp/schm_azspc] executing: git checkout -q origin/master
Using autogenlist config:
{
  "basePath": "recoveryservices/resource-manager",
  "namespace": "Microsoft.RecoveryServices"
}
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest --version=3.0.6320 --use=@autorest/azureresourceschema@3.0.92 --azureresourceschema --output-folder=/tmp/onfecptfi6 --multiapi --title=none /tmp/schm_azspc/specification/recoveryservices/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6247; node: v14.8.0, max-memory: 4096 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6320/node_modules/@autorest/core/dist' (3.0.6320)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.92->3.0.92)
Processing batch task - {"tag":"schema-recoveryservices-2016-06-01"} .
[5.6 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2016-06-01/Microsoft.RecoveryServices.json
Provider Namespace: Microsoft.RecoveryServices
API Version: 2016-06-01
Resource Types (Resource Group Scope):
- vaults
- vaults/certificates
- vaults/extendedInformation
================================================================================================================================
```
```
> azure-schema-generator@1.0.0 generate-single /home/leni/workspace/azure-resource-manager-schemas/generator
> ts-node cmd/generatesingle "recoveryservicesbackup/resource-manager"

[/tmp/schm_azspc] executing: git fsck
dangling commit f8bf79571777ef3f7a6ff365d81883ba08d37abc
[/tmp/schm_azspc] executing: git reset -q .
[/tmp/schm_azspc] executing: git checkout -q -- .
[/tmp/schm_azspc] executing: git clean -q -fd
[/tmp/schm_azspc] executing: git gc
[/tmp/schm_azspc] executing: git fetch -q
[/tmp/schm_azspc] executing: git checkout -q origin/master
Using autogenlist config:
{
  "basePath": "recoveryservicesbackup/resource-manager",
  "namespace": "Microsoft.RecoveryServices",
  "suffix": "Backup"
}
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest --version=3.0.6320 --use=@autorest/azureresourceschema@3.0.92 --azureresourceschema --output-folder=/tmp/cwyhlgsee4c --multiapi --title=none /tmp/schm_azspc/specification/recoveryservicesbackup/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6247; node: v14.8.0, max-memory: 4096 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6320/node_modules/@autorest/core/dist' (3.0.6320)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.92->3.0.92)
Processing batch task - {"tag":"schema-recoveryservices-2020-07-01"} .
Processing batch task - {"tag":"schema-recoveryservices-2020-02-02"} .
Processing batch task - {"tag":"schema-recoveryservices-2019-06-15"} .
Processing batch task - {"tag":"schema-recoveryservices-2019-05-13"} .
Processing batch task - {"tag":"schema-recoveryservices-2018-12-20"} .
Processing batch task - {"tag":"schema-recoveryservices-2017-07-01"} .
Processing batch task - {"tag":"schema-recoveryservices-2016-12-01"} .
Processing batch task - {"tag":"schema-recoveryservices-2016-08-10"} .
Processing batch task - {"tag":"schema-recoveryservices-2016-06-01"} .
[47 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2016-06-01/Microsoft.RecoveryServices.Backup.json
Provider Namespace: Microsoft.RecoveryServices
API Version: 2016-06-01
Resource Types (Resource Group Scope):
- vaults/backupFabrics/protectionContainers/protectedItems
- vaults/backupPolicies
================================================================================================================================
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2016-12-01/Microsoft.RecoveryServices.Backup.json
Provider Namespace: Microsoft.RecoveryServices
API Version: 2016-12-01
Resource Types (Resource Group Scope):
- vaults/backupFabrics/protectionContainers
- vaults/backupstorageconfig
================================================================================================================================
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2017-07-01/Microsoft.RecoveryServices.Backup.json
Provider Namespace: Microsoft.RecoveryServices
API Version: 2017-07-01
Resource Types (Resource Group Scope):
- vaults/backupFabrics/backupProtectionIntent
================================================================================================================================
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2019-05-13/Microsoft.RecoveryServices.Backup.json
Provider Namespace: Microsoft.RecoveryServices
API Version: 2019-05-13
Resource Types (Resource Group Scope):
- vaults/backupFabrics/protectionContainers/protectedItems
- vaults/backupPolicies
================================================================================================================================
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2019-06-15/Microsoft.RecoveryServices.Backup.json
Provider Namespace: Microsoft.RecoveryServices
API Version: 2019-06-15
Resource Types (Resource Group Scope):
- vaults/backupconfig
- vaults/backupFabrics/protectionContainers/protectedItems
- vaults/backupPolicies
================================================================================================================================
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2020-02-02/Microsoft.RecoveryServices.Backup.json
Provider Namespace: Microsoft.RecoveryServices
API Version: 2020-02-02
Resource Types (Resource Group Scope):
- vaults/privateEndpointConnections
================================================================================================================================
```
```
> azure-schema-generator@1.0.0 generate-single /home/leni/workspace/azure-resource-manager-schemas/generator
> ts-node cmd/generatesingle "recoveryservicessiterecovery/resource-manager"

[/tmp/schm_azspc] executing: git fsck
dangling commit f8bf79571777ef3f7a6ff365d81883ba08d37abc
[/tmp/schm_azspc] executing: git reset -q .
[/tmp/schm_azspc] executing: git checkout -q -- .
[/tmp/schm_azspc] executing: git clean -q -fd
[/tmp/schm_azspc] executing: git gc
[/tmp/schm_azspc] executing: git fetch -q
[/tmp/schm_azspc] executing: git checkout -q origin/master
Using autogenlist config:
{
  "basePath": "recoveryservicessiterecovery/resource-manager",
  "namespace": "Microsoft.RecoveryServices",
  "suffix": "SiteRecovery"
}
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest --version=3.0.6320 --use=@autorest/azureresourceschema@3.0.92 --azureresourceschema --output-folder=/tmp/elwxddi32nd --multiapi --title=none /tmp/schm_azspc/specification/recoveryservicessiterecovery/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6247; node: v14.8.0, max-memory: 4096 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6320/node_modules/@autorest/core/dist' (3.0.6320)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.92->3.0.92)
Processing batch task - {"tag":"schema-recoveryservices-2018-07-10"} .
Processing batch task - {"tag":"schema-recoveryservices-2018-01-10"} .
Processing batch task - {"tag":"schema-recoveryservices-2016-08-10"} .
[143.04 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2016-08-10/Microsoft.RecoveryServices.SiteRecovery.json
Provider Namespace: Microsoft.RecoveryServices
API Version: 2016-08-10
Resource Types (Resource Group Scope):
- vaults/replicationAlertSettings
- vaults/replicationFabrics
- vaults/replicationFabrics/replicationNetworks/replicationNetworkMappings
- vaults/replicationFabrics/replicationProtectionContainers
- vaults/replicationFabrics/replicationProtectionContainers/replicationProtectedItems
- vaults/replicationFabrics/replicationProtectionContainers/replicationProtectionContainerMappings
- vaults/replicationFabrics/replicationStorageClassifications/replicationStorageClassificationMappings
- vaults/replicationFabrics/replicationvCenters
- vaults/replicationPolicies
- vaults/replicationRecoveryPlans
================================================================================================================================
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2018-01-10/Microsoft.RecoveryServices.SiteRecovery.json
Provider Namespace: Microsoft.RecoveryServices
API Version: 2018-01-10
Resource Types (Resource Group Scope):
- vaults/replicationAlertSettings
- vaults/replicationFabrics
- vaults/replicationFabrics/replicationNetworks/replicationNetworkMappings
- vaults/replicationFabrics/replicationProtectionContainers
- vaults/replicationFabrics/replicationProtectionContainers/replicationMigrationItems
- vaults/replicationFabrics/replicationProtectionContainers/replicationProtectedItems
- vaults/replicationFabrics/replicationProtectionContainers/replicationProtectionContainerMappings
- vaults/replicationFabrics/replicationRecoveryServicesProviders
- vaults/replicationFabrics/replicationStorageClassifications/replicationStorageClassificationMappings
- vaults/replicationFabrics/replicationvCenters
- vaults/replicationPolicies
- vaults/replicationRecoveryPlans
================================================================================================================================
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2018-07-10/Microsoft.RecoveryServices.SiteRecovery.json
Provider Namespace: Microsoft.RecoveryServices
API Version: 2018-07-10
Resource Types (Resource Group Scope):
- vaults/replicationAlertSettings
- vaults/replicationFabrics
- vaults/replicationFabrics/replicationNetworks/replicationNetworkMappings
- vaults/replicationFabrics/replicationProtectionContainers
- vaults/replicationFabrics/replicationProtectionContainers/replicationMigrationItems
- vaults/replicationFabrics/replicationProtectionContainers/replicationProtectedItems
- vaults/replicationFabrics/replicationProtectionContainers/replicationProtectionContainerMappings
- vaults/replicationFabrics/replicationRecoveryServicesProviders
- vaults/replicationFabrics/replicationStorageClassifications/replicationStorageClassificationMappings
- vaults/replicationFabrics/replicationvCenters
- vaults/replicationPolicies
- vaults/replicationRecoveryPlans
- vaults/replicationVaultSettings
================================================================================================================================
```

### CODE REVIEW

File: generator/generate.ts [hunk 1]
```diff
@@ -139,7 +139,6 @@ async function generateSchema(readme: string, tmpFolder: string) {
         `--output-folder=${tmpFolder}`,
         `--multiapi`,
         '--title=none',
-        '--pass-thru:subset-reducer',
```
Comment (anthony-c-martin) [inline generator/generate.ts:142]: Is it safe to remove this? It was added as we had some issues with automatic generation of certain RPs.

File: generator/constants.ts [hunk 2]
```diff
@@ -29,8 +29,8 @@ export const managementGroupRootSchema = {
 };
 export const generatedSchemasTemplatePath = path.join(__dirname, 'resources/autogeneratedResources_template.json');
 
-export const autorestCoreVersion = '3.0.6274';
-export const azureresourceschemaVersion = '3.0.79';
+export const autorestCoreVersion = '3.0.6320';
+export const azureresourceschemaVersion = '3.0.92';
```
Comment (anthony-c-martin) [inline generator/constants.ts:33]: Is this needed for this PR? If not, possible to submit this as a separate change?

File: generator/autogenlist.ts [hunk 3]
```diff
@@ -454,6 +454,20 @@ const autogenlist: AutogenlistConfig[] = [
...
+    },
+    {
+        basePath: 'recoveryservicesbackup/resource-manager',
+        namespace: 'Microsoft.RecoveryServices',
+        suffix: "Backup"
+    },
+    {
+        basePath: "recoveryservices/resource-manager",
+        namespace: "Microsoft.RecoveryServices"
+    },
```
Comment (anthony-c-martin) [inline generator/autogenlist.ts:470]: Will the PR-driven swagger pipeline you added respect this file?

File: generator/generate.ts (hunk 1 above)
Comment (anthony-c-martin) [reply generator/generate.ts:142]: Ah, this was fixed by https://github.com/Azure/autorest.azureresourceschema/issues/52? If so, feel free to resolve this. 

Please could you test `npm run generate-single -- resources/resource-manager` and make sure it still works after this change?

**Comment by anthony-c-martin** [comment]
@leni-msft mind taking a look over my comments?

File: generator/autogenlist.ts (hunk 3 above)
Comment (leni-msft) [reply generator/autogenlist.ts:470]: yes, it will

File: generator/constants.ts (hunk 2 above)
Comment (leni-msft) [reply generator/constants.ts]: I will regenerate the PR, since we start use `--multiapi` to generate schemas

File: generator/generate.ts (hunk 1 above)
Comment (leni-msft) [reply generator/generate.ts:142]: This parameter turn out to be still needed. I'll recreate the PR.

**Comment by leni-msft** [comment]
replaced with #1451

**Pull request closed without merging**